src/Raven.Client/Documents/Session/Tokens/DeclareToken.cs

```diff
@@ -5,39 +5,50 @@ namespace Raven.Client.Documents.Session.Tokens
 {
     public class DeclareToken : QueryToken
     {
+        public enum DeclarationType
```

Comment:
Let's remove that and create maybe two methods.

`CreateFunction` and `CreateTimeSeries`